classes/Tools.php

```diff
@@ -3158,11 +3158,17 @@ public static function clearSf2Cache($env = null)
             $env = _PS_MODE_DEV_ ? 'dev' : 'prod';
         }
 
-        $dir = _PS_ROOT_DIR_ . '/var/cache/' . $env . '/';
-
-        register_shutdown_function(function() use ($dir) {
-            $fs = new Filesystem();
-            $fs->remove($dir);
+        register_shutdown_function(function() use ($env) {
```

Comment:
It looks like a wrong implementation for a wrong solution to a wrong problem... this will probably create weird bugs later, when people will think that when we clear the cache, we actually clear the cache (that we don't do here anymore, am I right?).

The problem with installation is a concurrency problem, this should be fixed in the installation process, making processes that need to be synchronous and ordered... synchronous or ordered.

If Smarty is creating files when Symfony is clearing cache, this means that during the cache clear, Smarty is still creating his cache: why do we create Smarty Cache while we clear cache just after that?

Sounds like a time-consuming and useless step, isn't it?

Reply:
> The fix needs to be done in the installation process, not in the core.

This is also impacting the core. Why do you want to remove files which are not concerned by that you want to delete?!

Reply:
> This is also impacting the core. Why do you want to remove files which are not concerned by that you want to delete?!

Maybe smarty cache and related files shouldn't be moved to Symfony cache folder at the first place?

Reply:
> But since a few month the installer is clearing the whole cache like an idiot rough several times.
Then, let's work on that! Why the cache is cleared more than one time, where and can we remove this behavior?